viewing_party/party.py

```diff
@@ -1,23 +1,169 @@
+        try:
+
+            most_watched = {}
+            description = []
+            for movie in user_data["watched"]:
+                if movie['genre']:
+                    description.append(movie['genre'])
+            for genre in description:
+                if genre in most_watched:
+                    most_watched[genre] += 1
+                else:
+                    most_watched[genre] = 1
+            most_popular = max(most_watched.values())
+            for keys, values in most_watched.items():
+                if values == most_popular:
+                    return keys
+                return user_data
+        except None:
+            return  None
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
 
-        
+def create_friends_dictionary(user_data):
+    friends_movies = {}
+    friends_movies[1] = user_data["friends"][0]
+    friends_movies[2] = user_data["friends"][1]
+    return friends_movies
+
+def get_unique_watched(user_data):
+    friends_movies = create_friends_dictionary(user_data)
+    users_unique_movies = []
+
+    for title in user_data["watched"]:
+        if title not in friends_movies[1]["watched"] and title not in friends_movies[2]["watched"]:
+            users_unique_movies.append(title)
+    return users_unique_movies
+
+def get_friends_unique_watched(user_data):
+    friends_unique_movies = []
+    friends_movies = create_friends_dictionary(user_data)
+    
+    for movie in friends_movies[1]["watched"]:
+        if movie not in user_data["watched"]:
+            if movie not in friends_unique_movies:
+                friends_unique_movies.append(movie)
+    for friend_movies in friends_movies[2]["watched"]:
+        if friend_movies not in user_data["watched"]:
+            if friend_movies not in friends_unique_movies:
+                friends_unique_movies.append(friend_movies)
+    return friends_unique_movies
+
 # -----------------------------------------
 # ------------- WAVE 4 --------------------
 # -----------------------------------------
+def get_available_recs(user_data):
+    recommendations = []
+    friends_movies = create_friends_dictionary(user_data)
+
+    for movie in friends_movies[1]["watched"]:
+        if movie not in user_data["watched"] and movie["host"] in user_data["subscriptions"]:
+            recommendations.append(movie)
+        for movie in friends_movies[2]["watched"]:
+            if movie not in user_data["watched"] and movie["host"] in user_data["subscriptions"]:
+                recommendations.append(movie)
+    return recommendations
 
 # -----------------------------------------
 # ------------- WAVE 5 --------------------
 # -----------------------------------------
 
+def get_new_rec_by_genre(user_data):
+    recommendations = []
+    friends_movies = create_friends_dictionary(user_data)
+
+    if user_data["watched"] != []:
+        if friends_movies[1]["watched"] != []:

```

Comment:
We don't need to check whether a list is empty before trying to iterate over it. If we iterate over an empty list, the body of the loop will simply not run. But there's no error. We can often structure our code to take advantage of this by initialing a variable to whatever the expected result we want if there isn't any data (often an empty list, `0` or `None`), but updating it within the loop (such as appending to a list, adding to a number, or assigning a found reference).